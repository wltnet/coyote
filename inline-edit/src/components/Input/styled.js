import styled from "styled-components";

export const Label = styled.label`
`;

export const InputField = styled.input`
  font-size: 1rem;
`;
