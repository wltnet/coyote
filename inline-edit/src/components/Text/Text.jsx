import React from "react";
import Icon from "../Icon";
import { INPUT_STATUS } from "../../util/inputStatus";
import { Error, Wrapper, InputText } from "./styled";

const Text = ({ text, handleClick, status }) => (
<>
  <Wrapper onClick={handleClick}>
    <InputText>{text}</InputText>
    <Icon status={status} />
  </Wrapper>
  {status === INPUT_STATUS.ERROR && <Error>Oops! Something has gone terribly wrong!</Error> }
</>
);
export default Text;
