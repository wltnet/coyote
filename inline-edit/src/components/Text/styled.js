import styled from "styled-components";

export const Wrapper = styled.div`
  display: flex;
  justify-content: space-between;
`;

export const InputText = styled.div`
`;

export const Error = styled.div`
  font-size: 0.8rem;
  color: red;
`;
