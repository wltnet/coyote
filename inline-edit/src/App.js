import React, { useState } from 'react';
import Text from "./components/Text";
import Input from "./components/Input";
import { INPUT_STATUS } from "./util/inputStatus";
import { saveTextSuccess } from "./util/mockPromise";
import './App.css';

const inputField = {
  label: "helloInput",
  name: "helloInput",
}

const App = () => {
  const [text, setText] = useState("Hello World");
  const [inputText, setInputText] = useState(text);
  const [edit, setEdit] = useState(false);
  const [status, setStatus] = useState();

  const udpateText = async (inputValue) => {
    setEdit(false);

    // No need to save if there is no change.
    if (text === inputValue) return;

    const prevText = text;

    setStatus(INPUT_STATUS.LOADING);
    setText(inputValue);

    // There is 50:50 chance success rate
    try {
      await saveTextSuccess(500, inputValue);
      setStatus(INPUT_STATUS.SUCCESS);
    } catch (error) {
      setStatus(INPUT_STATUS.ERROR);
      setInputText(prevText);
      setText(prevText);
    }

    setTimeout(() => { setStatus(INPUT_STATUS.SAVED); }, 2000);
  }

  const handleClick = () => {
    setEdit(true);
  }

  const handleBlur = (ev) => {
    udpateText(ev.target.value)
  }

  const handleKeyDown = (ev) => {
    if (ev.key === 'Enter') {
      udpateText(ev.target.value)
    }
  }

  const handleChange = (ev) => {
    setInputText(ev.target.value);
  }

  return (
    <div className="App">
      <div className="Wrapper">
        {edit ? 
          <Input
            inputField={inputField}
            value={inputText}
            handleBlur={handleBlur}
            handleChange={handleChange}
            handleKeyDown={handleKeyDown}
          /> : 
          <Text
            text={text}
            handleClick={handleClick}
            status={status}
          />}
      </div>
    </div>
  );
}

export default App;
