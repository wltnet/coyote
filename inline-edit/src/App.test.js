import React from 'react';
import { shallow } from "enzyme";
import App from './App';
import Text from "./components/Text";
import { Wrapper } from "./components/Text/styled";

jest.mock("./components/Input", () => "Input");

describe("<App />", () => {
  it("should match the snapshot when first render", () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should match the snapshot after static text has been click", () => {
    const wrapper = shallow(<App />);

    wrapper.find(Text).dive().find(Wrapper).simulate("click");
    expect(wrapper).toMatchSnapshot();
  });
});