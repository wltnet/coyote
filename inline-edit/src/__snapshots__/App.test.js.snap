// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should match the snapshot 1`] = `
<div
  className="App"
>
  <div
    className="Wrapper"
  >
    <Text
      handleClick={[Function]}
      text="Hello World"
    />
  </div>
</div>
`;

exports[`<App /> should match the snapshot 2`] = `
<div
  className="App"
>
  <div
    className="Wrapper"
  >
    <Input
      handleBlur={[Function]}
      handleChange={[Function]}
      handleKeyDown={[Function]}
      inputField={
        Object {
          "label": "helloInput",
          "name": "helloInput",
        }
      }
      value="Hello World"
    />
  </div>
</div>
`;

exports[`<App /> should match the snapshot after static text has been click 1`] = `
<div
  className="App"
>
  <div
    className="Wrapper"
  >
    <Input
      handleBlur={[Function]}
      handleChange={[Function]}
      handleKeyDown={[Function]}
      inputField={
        Object {
          "label": "helloInput",
          "name": "helloInput",
        }
      }
      value="Hello World"
    />
  </div>
</div>
`;

exports[`<App /> should match the snapshot when first render 1`] = `
<div
  className="App"
>
  <div
    className="Wrapper"
  >
    <Text
      handleClick={[Function]}
      text="Hello World"
    />
  </div>
</div>
`;
